src/main/java/duke/util/TaskList.java

```diff
@@ -66,6 +66,21 @@ public void deleteTask(int index) {
         taskList.remove(index - 1);
     }
 
+    /**
+     * Checks if the description of a task given by index contains a given filter string.
+     * If the index is invalid, this method will throw an assertion error. This search is
+     * case-insensitive.
+     *
+     * @param index     1-indexed value indicating position of the task to check
+     * @param filter    the string to check
+     * @return          true if task description contains filter string
+     */
+    public boolean taskDescriptionContains(int index, String filter) {
```

Comment:
Would it be possible to use an existing method (eg. containsIgnoreCase) instead of defining another method?

Reply:
Did not know about that! Thank you for the suggestion.

Reply:
On second thought, it seems quite frivolous to include an external library for a single function. Since I want to keep this function in `TaskList` for abstraction purposes, I feel that leaving it as it is should be fine.